### PR TITLE
Update KafkaChannel controller to use genreconciler

### PR DESCRIPTION
**Pull request by lberk**
----
# KafkaChannel controller updated to use the +genreconciler implementation

### CODE REVIEW

**Comment by matzew** [comment]
```
kafka/channel/pkg/reconciler/controller/kafkachannel_test.go:342:4: cannot use &Reconciler literal (type *Reconciler) as type controller.Reconciler in return argument:
	*Reconciler does not implement controller.Reconciler (missing Reconcile method)
kafka/channel/pkg/reconciler/controller/kafkachannel_test.go:406:4: cannot use &Reconciler literal (type *Reconciler) as type controller.Reconciler in return argument:
	*Reconciler does not implement controller.Reconciler (missing Reconcile method)
kafka/channel/pkg/reconciler/controller/kafkachannel_test.go:474:4: cannot use &Reconciler literal (type *Reconciler) as type controller.Reconciler in return argument:
	*Reconciler does not implement controller.Reconciler (missing Reconcile method)
```

from the log

File: kafka/channel/pkg/reconciler/controller/kafkachannel.go [hunk 1]
```diff
@@ -46,24 +42,17 @@ import (
 	"knative.dev/eventing/pkg/reconciler"
 	"knative.dev/eventing/pkg/reconciler/names"
 	"knative.dev/pkg/apis"
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel.go]: Format Go code:
```suggestion
	"knative.dev/pkg/apis"

```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion
	"k8s.io/apimachinery/pkg/runtime"
	eventingClient "knative.dev/eventing/pkg/client/injection/client"
```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ import (
...
 	"knative.dev/eventing/pkg/reconciler"
 	"knative.dev/eventing/pkg/utils"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	eventingClient "knative.dev/eventing/pkg/client/injection/client"
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion

```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 4]
```diff
@@ -340,7 +342,10 @@ func TestAllCases(t *testing.T) {
 			endpointsLister:      listers.GetEndpointsLister(),
 			kafkaClusterAdmin:    &mockClusterAdmin{},
 			kafkaClientSet:       fakekafkaclient.Get(ctx),
+			KubeClientSet:        kubeclient.Get(ctx),
+			EventingClientSet: eventingClient.Get(ctx),
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion
			EventingClientSet:    eventingClient.Get(ctx),
```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 5]
```diff
@@ -374,14 +379,15 @@ func TestTopicExists(t *testing.T) {
 			),
 		}},
 		WantEvents: []string{
-			Eventf(corev1.EventTypeNormal, channelReconciled, "KafkaChannel reconciled"),
+			Eventf(corev1.EventTypeNormal, "KafkaChannelReconciled", `KafkaChannel reconciled: "test-namespace/test-kc"`),
+
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion

```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 6]
```diff
@@ -404,7 +410,11 @@ func TestTopicExists(t *testing.T) {
 				},
 			},
 			kafkaClientSet: fakekafkaclient.Get(ctx),
+			KubeClientSet:        kubeclient.Get(ctx),
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion
			kafkaClientSet:    fakekafkaclient.Get(ctx),
			KubeClientSet:     kubeclient.Get(ctx),
```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 7]
```diff
@@ -404,7 +410,11 @@ func TestTopicExists(t *testing.T) {
...
+			EventingClientSet: eventingClient.Get(ctx),
+
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion

```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 8]
```diff
@@ -472,7 +482,11 @@ func TestDeploymentUpdatedOnImageChange(t *testing.T) {
 				},
 			},
 			kafkaClientSet: fakekafkaclient.Get(ctx),
+			KubeClientSet:        kubeclient.Get(ctx),
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion
			kafkaClientSet:    fakekafkaclient.Get(ctx),
			KubeClientSet:     kubeclient.Get(ctx),
```

File: kafka/channel/pkg/reconciler/controller/kafkachannel_test.go [hunk 9]
```diff
@@ -472,7 +482,11 @@ func TestDeploymentUpdatedOnImageChange(t *testing.T) {
...
+			EventingClientSet: eventingClient.Get(ctx),
+
```
Comment (mattmoor) [inline kafka/channel/pkg/reconciler/controller/kafkachannel_test.go]: Format Go code:
```suggestion

```

**Comment by lberk** [comment]
@n3wscott ping? Could I get a review on the move to the genreconciler? thanks!

**Pull request merged**